### ISSUE TITLE
Getting invoke from function pointers with reflection doesn't work anymore

### ISSUE THREAD

**Issue by Moneyl** [beefytech/Beef#1913]
I'm using reflection to get the `Invoke` method off of function pointers. Just updated to 9d72043a71fd18b79b051bc079e7f51c834ed394 and this doesn't work anymore. `GetMethod` doesn't find the invoke method. It doesn't work in comptime anymore either (using it for code generation).

I believe I was on 99dcf4f07538125dcc4bc3c46ac38bf457e5b5ea before this.

```cs
using System;
using System.Diagnostics;

namespace FunctionInvokeReflection;

[Reflect(.All), AlwaysInclude]
struct FunctionReflectionTest
{
    [Reflect(.All)]
    public function void(int a) test;
}

class Program
{
    public static int Main(String[] args)
    {
        for (var vtfunc in typeof(FunctionReflectionTest).GetFields())
        {
            if (vtfunc.FieldType.BaseType != typeof(Function))
                continue;
            
            var getFunc = vtfunc.FieldType.GetMethod("Invoke");
            if (getFunc case .Err)
            {
                //It used to find it even without the reflect attribute. Now it doesn't find it at all.
                Debug.WriteLine("Invoke not found!");
            }
        }

        return 0;
    }
}
```

[FunctionInvokeReflection.zip](https://github.com/beefytech/Beef/files/13763042/FunctionInvokeReflection.zip)

**Comment by bfiete** [comment]
Fixed at b98f1ea6a46e267eabd880c92b6fd5a364bf022e

**Comment by bfiete** [comment]
Your repo actually doesn't work on the build you specified. This fix doesn't either, but I think it will probably fix your actual issue.
